test/e2e/apps/replica_set.go

```diff
@@ -102,7 +102,14 @@ var _ = SIGDescribe("ReplicaSet", func() {
 		testReplicaSetServeImageOrFail(f, "private", privateimage.GetE2EImage())
 	})
 
-	ginkgo.It("should surface a failure condition on a common issue like exceeded quota", func() {
+	/*
+	   Release : v1.16
+	   Testname: Validate ReplicaSet pod limit restrictions
+	   Description: Apply quota to the namespace, prohibiting more than two pods.
+	   When creating a ReplicaSet of 3 of a pod, ReplicaSetConditions MUST include the failure of creating more than the quota allows.
+	   Replicas of pods will be scaled down to two MUST succeed, as the quota allows this amount of replicas to exist.
+	*/
+	framework.ConformanceIt("should surface a failure condition on a common issue like exceeded quota", func() {
 		testReplicaSetConditionCheck(f)
```

Comment:
The condition check https://github.com/kubernetes/kubernetes/pull/80148/files#diff-34aadb3c023dcf4d181ab896ce74d84fL225 is wrong. It should be checking that the cond.Status is true. This test as written will pass even if that condition is present but set to False.

Reply:
See https://github.com/kubernetes/kubernetes/pull/74891#issuecomment-479234581 which also asks for this to check that the condition does not occur when it shouldn't (so, say, change the spec to 2 replicas and the condition should clear up).

Reply:
> The condition check https://github.com/kubernetes/kubernetes/pull/80148/files#diff-34aadb3c023dcf4d181ab896ce74d84fL225 is wrong. It should be checking that the cond.Status is true. This test as written will pass even if that condition is present but set to False.

cond.Status appears to be an [string / api.ConditionStatus](https://github.com/kubernetes/kubernetes/blob/7f1ae0e32d2d42efc243b0b3d69a95797ce67046/pkg/apis/apps/types.go#L786), not sure if the comparison for the return value can be valid.

Reply:
> The condition check https://github.com/kubernetes/kubernetes/pull/80148/files#diff-34aadb3c023dcf4d181ab896ce74d84fL225 is wrong. It should be checking that the cond.Status is true. This test as written will pass even if that condition is present but set to False.

cond.Status appears to be an [string / api.ConditionStatus](https://github.com/kubernetes/kubernetes/blob/7f1ae0e32d2d42efc243b0b3d69a95797ce67046/pkg/apis/apps/types.go#L786), not sure if the return value would be valid if trying to return as bool.

Build log (relevant lines):
```
test/e2e/apps/replica_set.go:232:22: cannot convert true (type untyped bool) to type "k8s.io/kubernetes/vendor/k8s.io/api/core/v1".ConditionStatus
test/e2e/apps/replica_set.go:232:22: invalid operation: cond.Status == true (mismatched types "k8s.io/kubernetes/vendor/k8s.io/api/core/v1".ConditionStatus and bool)
```

Reply:
you can compare against these known values
https://github.com/kubernetes/kubernetes/blob/acaac181dc8f4c16ed0f5e47f687a6475cae49f3/staging/src/k8s.io/api/core/v1/types.go#L2304-L2308